WireAPI/Sources/WireAPI/APIs/FeatureConfigsAPI/FeatureConfigsAPIV0.swift

```diff
@@ -0,0 +1,164 @@
+
+    init(httpClient: HTTPClient) {
+        self.httpClient = httpClient
+    }
+
+    var apiVersion: APIVersion {
+        .v0
+    }
+
+    // MARK: - Get all feature configs
+
+    func getFeatureConfigs() async throws -> [FeatureConfig] {
+        let request = HTTPRequest(
+            path: "\(pathPrefix)/feature-configs",
+            method: .get
+        )
+
+        let response = try await httpClient.executeRequest(request)
+
+        return try ResponseParser()
+            .success(code: .ok, type: FeatureConfigsResponseAPIV0.self)
+            .failure(code: .forbidden, label: "operation-denied", error: FeatureConfigsAPIError.insufficientPermissions)
+            .failure(code: .forbidden, label: "no-team-member", error: FeatureConfigsAPIError.userIsNotTeamMember)
+            .failure(code: .notFound, label: "no-team", error: FeatureConfigsAPIError.teamNotFound)
+            .parse(response)
+    }
+
+}
+
+/// A namespace for all feature config responses
+
+enum FeatureConfigResponse {
+
+    struct AppLockV0: Decodable {
+
+        let enforceAppLock: Bool
+        let inactivityTimeoutSecs: UInt
+
+    }
+
+    struct ClassifiedDomainsV0: Decodable {
+
+        let domains: Set<String>
+
+    }
+
+    struct MLSV0: Decodable {
+
+        let protocolToggleUsers: Set<UUID>
+        let defaultProtocol: MessageProtocol
+        let allowedCipherSuites: [MLSCipherSuite]
+        let defaultCipherSuite: MLSCipherSuite
+
+    }
+
+    struct SelfDeletingMessagesV0: Decodable {
+
+        let enforcedTimeoutSeconds: UInt
+
+    }
+
+}
+
+struct FeatureConfigsResponseAPIV0: Decodable, ToAPIModelConvertible {
+
+    let appLock: FeatureWithConfig<FeatureConfigResponse.AppLockV0>
+    let classifiedDomains: FeatureWithConfig<FeatureConfigResponse.ClassifiedDomainsV0>
+    let conferenceCalling: FeatureWithoutConfig
+    let conversationGuestLinks: FeatureWithoutConfig
+    let digitalSignatures: FeatureWithoutConfig
+    let fileSharing: FeatureWithoutConfig
+    let selfDeletingMessages: FeatureWithConfig<FeatureConfigResponse.SelfDeletingMessagesV0>
+    let mls: FeatureWithConfig<FeatureConfigResponse.MLSV0>?
+
+    func toAPIModel() -> [FeatureConfig] {
+        var featureConfigs: [FeatureConfig] = []
+
+        let appLockConfig = AppLockFeatureConfig(
+            status: appLock.status,
+            isMandatory: appLock.config.enforceAppLock,
+            inactivityTimeoutInSeconds: appLock.config.inactivityTimeoutSecs
+        )
+
+        featureConfigs.append(.appLock(appLockConfig))
+
+        let classifiedDomainsConfig = ClassifiedDomainsFeatureConfig(
+            status: classifiedDomains.status,
+            domains: classifiedDomains.config.domains
+        )
+
+        featureConfigs.append(.classifiedDomains(classifiedDomainsConfig))
+
+        let conferenceCallingConfig = ConferenceCallingFeatureConfig(
+            status: conferenceCalling.status,
+            useSFTForOneToOneCalls: nil
```

Comment:
Is this a `Bool?`, perhaps we can make it non optional and default to `false`.

Reply:
yes, I figured since this property was introduced in v6, having a default to false could be misleading when this property doesn't actually exist for versions < v6 could introduce some issues, however looking at codebase we do have a public init that default this property to false, so I will make the change, thanks.

Reply:
done